### PR TITLE
Preview NASA-TLX and SMEQ slider values on hover

### LINKED ISSUE #1089: Hover effect for linear scale (TLX and SMEQ libraries)
On the TLX and SMEQ libraries, 

add a hover effect  (black tick instead of red) with a tool-tip number

Image

It's awkward without it. 

### Summary
TLX and SMEQ sliders currently show a tooltip at the committed red value when the track is hovered, rather than previewing the value beneath the pointer. Add a transient black tick and numeric tooltip that follow the pointer without changing the participant’s committed answer.

### Affected surface
- Users and trigger/reproduction: Participants hovering an enabled slider configured with `tlxStyle` or `smeqStyle`.
- Exact entry points: `/library-nasa-tlx`, `/library-smeq`, and imported TLX or SMEQ components in other studies.
- UI elements and states: Horizontal TLX scales, vertical SMEQ scales, committed red ticks, transient hover previews, pointer exit, and disabled/read-only replay states.
- Relevant contracts: Slider values, ranges, steps, snapping, answer updates, and persistence remain unchanged.

### Required behavior
When the pointer enters or moves across an enabled TLX or SMEQ track, display a black preview tick at the candidate value beneath the pointer and attach a tooltip showing that value. Calculate the preview using the same minimum, maximum, step, and snapping rules as an actual selection.

The committed red tick must remain at the current answer, and hovering alone must not invoke an answer change or write participant data. Remove the preview tick and tooltip when the pointer leaves the track. Existing click, drag, keyboard, and replay behavior must continue unchanged.

Disabled or read-only sliders must display their committed value without offering a hover preview.

### Acceptance criteria
- [ ] TLX sliders show a black hover tick and matching numeric tooltip at the pointer-derived horizontal value.
- [ ] SMEQ sliders show a black hover tick and matching numeric tooltip at the pointer-derived vertical value.
- [ ] Preview values are clamped to the configured range and follow the slider’s existing step or snapping rules.
- [ ] Hovering does not move the committed red tick, invoke the answer change handler, or modify participant data.
- [ ] The preview tick and tooltip disappear when the pointer leaves the track.
- [ ] The tooltip remains visible and correctly positioned at both range endpoints.
- [ ] Existing click and drag interactions continue to commit values normally.
- [ ] Disabled and replayed sliders retain their committed indicator without showing an alternative hover value.
- [ ] Standard sliders without `tlxStyle` or `smeqStyle` retain their existing behavior.
- [ ] Existing labels, marks, layout, keyboard interaction, and accessible value semantics do not regress.

### Tests and verification
- Add component tests for TLX and SMEQ pointer previews at the minimum, middle, and maximum positions; step/snapping behavior; pointer exit; disabled state; and the absence of answer updates during hover.
- Add Playwright coverage in Chromium using `/library-nasa-tlx` and `/library-smeq` to verify the visible preview, tooltip value, committed red value, and unchanged answer until selection.

### Non-goals and assumptions
- No new Study Config properties, schemas, dependencies, storage fields, or persistence behavior.
- No behavior changes for standard sliders.
- A keyboard equivalent for the transient pointer preview and broader SMEQ accessibility changes are outside this issue; existing interaction and accessibility behavior must not regress.

### PR DESCRIPTION
**Pull request by jaykim1213**
### Does this PR close any open issues?
Closes #1089

### Give a longer description of what this PR addresses and why it's needed

Hover effect for the sliders
- NASA-TLX
- SMEQ

### Provide pictures/videos of the behavior before and after these changes (optional)
- NASA-TLX
image

- SMEQ
Screenshot 2026-08-24 at 11 25 37 AM

### Are there any additional TODOs before this PR is ready to go?
TODOs:
- [ ] Update relevant documentation
- [ ] ...

### CODE REVIEW

**Comment by jaykim1213** [comment]
@codex

File: src/components/response/tests/ResponseInput.spec.tsx [hunk 1]
```diff
@@ -195,9 +197,13 @@ vi.mock('@mantine/hooks', () => ({
   useMove: vi.fn(() => ({ ref: () => undefined, active: false })),
 }));
 
-vi.mock('../sliderBreaks', () => ({
-  generateSliderBreakValues: vi.fn(() => []),
-}));
+vi.mock('../sliderBreaks', async (importOriginal) => {
+  const actual = await importOriginal<typeof import('../sliderBreaks')>();
```
[the remark this answers is not in the document]
Comment (jaykim1213) [reply src/components/response/tests/ResponseInput.spec.tsx]: Updated

File: src/components/response/sliderBreaks.ts [hunk 2]
```diff
@@ -37,3 +37,33 @@ export function generateSliderBreakValues(min: number, max: number, spacing?: nu
...
+
+  if (snapValues?.length) {
+    return snapValues.reduce((closest, value) => (
+      Math.abs(value - rawValue) < Math.abs(closest - rawValue) ? value : closest
+    ));
+  }
+
+  const stepSize = step ?? (max - min) / 100;
+  // Extra guard for JavaScript floating point precision issues when calculating the number of decimal places
+  const precision = stepSize.toString().split('.')[1]?.length ?? 0;
```
[the remark this answers is not in the document]
Comment (jaykim1213) [reply src/components/response/sliderBreaks.ts]: Fixed by calculating precision from both the minimum and the step

**Review by JackWilb — APPROVED** [review]
Verified the exact current head manually: NASA-TLX sliders use step=5, pointer selection resolves to five-point increments, and keyboard movement advances by five. The existing head already implements the requested behavior; no code change was needed.
